dask/dataframe/io/parquet/arrow.py

```diff
@@ -1400,143 +1376,110 @@ def _process_metadata(
-                    # Get full path (empty strings should be ignored)
-                    full_path = fs.sep.join(
-                        [p for p in [data_path, filename] if p != ""]
-                    )
-                    pkeys = partition_keys.get(full_path, None)
-                    if partition_obj and pkeys is None:
-                        continue  # This partition was filtered
-                    part = {
-                        "piece": (
-                            frag_map[(full_path, rg_list[0])]
-                            if pass_frags
-                            else full_path,
-                            rg_list,
-                            pkeys,
-                        ),
-                    }
-                    parts.append(part)
-                    if gather_statistics:
-                        stat = cls._aggregate_stats(
-                            filename,
-                            file_row_group_stats[filename][i:i_end],
-                            file_row_group_column_stats[filename][i:i_end],
-                            stat_col_indices,
-                        )
-                        stats.append(stat)
-        else:
-            for filename, row_groups in file_row_groups.items():
-                # Get full path (empty strings should be ignored)
-                full_path = fs.sep.join([p for p in [data_path, filename] if p != ""])
-                pkeys = partition_keys.get(full_path, None)
-                if partition_obj and pkeys is None:
-                    continue  # This partition was filtered
-                part = {
-                    "piece": (
-                        frag_map[(full_path, row_groups[0])]
-                        if pass_frags
-                        else full_path,
-                        row_groups,
-                        pkeys,
-                    ),
-                }
-                parts.append(part)
-                if gather_statistics:
-                    stat = cls._aggregate_stats(
-                        filename,
-                        file_row_group_stats[filename],
-                        file_row_group_column_stats[filename],
-                        stat_col_indices,
-                    )
-                    stats.append(stat)
+        return (
+            file_row_groups,
+            file_row_group_stats,
+            file_row_group_column_stats,
+            gather_statistics,
+            frag_map,
+        )
 
-        common_kwargs = {
-            "partitioning": partition_info["partitioning"],
-            "partitions": partition_obj,
-            "categories": categories,
-            "filters": filters,
-            "schema": schema,
-        }
+    @classmethod
+    def _make_part(
+        cls,
+        filename,
+        rg_list,
+        fs=None,
+        partition_keys=None,
+        partition_obj=None,
+        data_path=None,
+        frag_map=None,
+    ):
```

Comment:
Used by both pyarrow engines as the `make_part_func` argument to `_row_groups_to_parts`.  The fastparquet engine has its own definition.